File: Implementation/DataStructures/java/ArrayStack.java
class Stack {
    public static int arrStack[], currentTopPos, stackSize;

    Stack(int stackCapacity) {
        arrStack = new int[stackCapacity + 1];
        currentTopPos = 0;
        stackSize = stackCapacity;
    }

    public static boolean isEmpty() {
        return currentTopPos == 0;
    }

    public static boolean isFull() {
        return currentTopPos == stackSize - 1;
    }

    public static int push(int newData) {
        if (isFull())
            return Integer.MAX_VALUE; // stack overflow

        ++currentTopPos;
        arrStack[currentTopPos] = newData;
        return arrStack[currentTopPos];
    }

    public static int pop() {
        if (isEmpty())
            return Integer.MIN_VALUE; // stack underflow

        int poppedData = arrStack[currentTopPos];
        arrStack[currentTopPos] = 0;
        --currentTopPos;

        return poppedData;
    }

    public static int top() {
        if (isEmpty())
            return Integer.MIN_VALUE; // stack underflow

        return arrStack[currentTopPos];
    }
}

public class ArrayStack {
    public static void main(String[] args) {
        Stack testStack = new Stack(4);

        testStack.push(10);
        testStack.push(824);
        testStack.push(923);
        testStack.push(9614);

        System.out.println(testStack.pop());

        testStack.push(3659);

        System.out.println(testStack.isEmpty() ? "empty" : "not empty");
        System.out.println(testStack.isFull() ? "full" : "not full");

        testStack.pop();

        System.out.println(testStack.top());
    }
}

File: Implementation/DataStructures/python/ArrayStack.py
import sys

class Stack:
    def __init__(self, stack_capacity):
        self.stack = [0] * (stack_capacity + 1)
        self.stack_size = stack_capacity
        self.current_top_pos = 0

    def is_empty(self):
        return self.current_top_pos == 0

    def is_full(self):
        return self.current_top_pos == (self.stack_size - 1)

    def push(self, new_data):
        if self.is_full():
            return sys.maxsize # stack overflow

        self.current_top_pos = self.current_top_pos + 1;
        self.stack[self.current_top_pos] = new_data

        return self.stack[self.current_top_pos]

    def pop(self):
        if self.is_empty():
            return (-sys.maxsize) # stack underflow

        popped = self.stack[self.current_top_pos];
        self.stack[self.current_top_pos] = 0
        self.current_top_pos = self.current_top_pos - 1;

        return popped

    def top(self):
        if self.is_empty():
            return (-sys.maxsize) # stack underflow

        return self.stack[self.current_top_pos]

test_stack = Stack(4)

test_stack.push(10)
test_stack.push(52)
test_stack.push(238)
test_stack.push(528)
test_stack.push(962)

print(test_stack.pop())

if test_stack.is_empty():
    print("empty")
else:
    print("not empty")

if test_stack.is_full():
    print("full")
else:
    print("not full")

print(test_stack.top())
